keras/engine/topology.py

```diff
@@ -1672,17 +1672,20 @@ def build_map_of_graph(tensor, seen_nodes=set(), depth=0,
                 layers_by_depth[depth] = []
             layers_by_depth[depth].append(layer)
 
+        # get sorted list of node depths
+        # -- this may be a superset of layer depths
         depth_keys = list(nodes_by_depth.keys())
         depth_keys.sort(reverse=True)
 
         # set self.layers and self.layers_by_depth
         layers = []
         for depth in depth_keys:
-            layers_for_depth = layers_by_depth[depth]
-            # container.layers needs to have a deterministic order
-            layers_for_depth.sort(key=lambda x: x.name)
-            for layer in layers_for_depth:
-                layers.append(layer)
+            if depth in layers_by_depth:
```

Comment:
Ok, that's actually how I originally fixed it (i.e. collect sorted key lists for both layers and nodes), but I went for the version with fewer lines of code instead....